### ISSUE TITLE
Bad filters ignored on POST requests

### ISSUE THREAD

**Issue by ThuGapy** [PostgREST/postgrest#3562]
Currently, POST requests parameters seem to be fetched from the body. However, some HTTP integrations with 3rd party tools appends query parameters to the URL even if it is a POST request. As far as I can understand, currently, PostgREST will try to parse the query parameter as a filter instead of simply ignoring it, causing the POST request to return a HTTP error code of 400 (Bad Request).

One of the suggested solution is to use a reverse proxy like nginx to remove the query parameter. The problem with that is that it creates edge-cases and it is frankly a complicated solution to a simple problem.

My suggestion would be to simply ignore the bad filters on POST requests instead of returning 400. This would simplify integration with 3rd party tools significantly and eliminate the need of a reverse proxy for something as basic as a useless query parameter.

If ignoring the bad filters on POST requests isn't something that is feasable, maybe being able to pass parameters from the URL to the PL/pgSQL function on a POST request could be an interesting solution.

**Comment by ThuGapy** [comment]
> I don't think we want to do that. Silently ignoring stuff is not helpful, e.g. when mistyping things etc. I agree, but to me, how it is currently is probably the worst possible way when were are talking about using PostgREST for integrations where we have no control over how the data is being sent.
> maybe being able to pass parameters from the URL to the PL/pgSQL function on a POST request could be an interesting solution.
> That's better. But what would happen if you pass the same parameter in the body **and** in the URL?
To me, how it's currently done where POST request try to get the parameters from the body is correct, getting parameters from the URL could be a fallback in case it's not found in the body. Realisticly, in an ideal world, most if not all data should be passed in the body, so I wouldn't have a problem with query parameters as fallback.

**Comment by ThuGapy** [comment]
Yeah, I would say I agree with your idea, we get the best of both worlds and it covers most cases

**Comment by tjoekbezoer** [comment]
I agree that being able to pass parameters through the querystring would be very convenient.

Regarding the ambiguity when passing the same parameter via QS and body; wouldn't it be acceptable to throw an error in this case? Since it very likely is...

It has the added benefit of not further complicating the interpretation of function signatures.

**Comment by steve-chavez** [comment]
> CREATE FUNCTION rpc(json, query_arg text) ...
> pass query_arg from the query string

Would this be the full query string or just a single `<query_arg>` in `query_key=<query_arg>` ? I assume the latter would be the most useful one as users wouldn't have to parse the querystring. It would be like [Nginx's $arg_](https://nginx.org/en/docs/http/ngx_http_core_module.html#var_arg_). In that case we would have multiple parameters in the function.

 Also it wouldn't consider filters like `eq.val` right? It would only take function parameters?

**Comment by wolfgangwalther** [comment]
My proposal was meant to work exactly the same way as named arguments work for `GET` + query string. Multiple named arguments would map to multiple query params.

**Comment by tjoekbezoer** [comment]
The downside to separating query string parameters from body parameters is that the body parameters are less specifically defined than the querystring parameters (with the first argument being a catch-all json object).

From my perspective it would be the most flexible if all of a function's parameters could be individually defined, and be passed in through either the querystring or the body. If they exist in both an error is thrown because of ambiguity.
